Fix invalid escape sequences in docstrings

tchamberlin opened the pull request:
These were resulting in SyntaxWarning in Python 3.12:

```plain
dysh/src/dysh/plot/specplot.py:10: SyntaxWarning: invalid escape sequence '\*'
  """
dysh/src/dysh/plot/specplot.py:91: SyntaxWarning: invalid escape sequence '\*'
  """
dysh/src/dysh/plot/specplot.py:167: SyntaxWarning: invalid escape sequence '\*'
  """Set x and y labels according to spectral units
dysh/src/dysh/spectra/scan.py:155: SyntaxWarning: invalid escape sequence '\d'
  """Average all polarizations in all scans in this ScanBlock
dysh/src/dysh/spectra/scan.py:175: SyntaxWarning: invalid escape sequence '\d'
  """Average all times and polarizations in all scans this ScanBlock
dysh/src/dysh/spectra/scan.py:697: SyntaxWarning: invalid escape sequence '\d'
  """
```

mpound commented:
Yeah, these were to have proper **kwargs shown in documentation.  I'll accept this change and we can see what it does to docs.

mpound reviewed (changes requested):
The double \ in the :math: lines is incorrect. These are latex symbols with a single \

tchamberlin commented:
Ah okay! I've reverted all of this and changed to raw-string-literal docstrings. These are already being used elsewhere, and the numpy examples use it as well: https://github.com/numpy/numpydoc/blob/main/doc/example.py

mpound reviewed (approved):
Also, FYI the r at the beginning of some docstrings allows one to use latex math markup in the documentation itself.
